### PR TITLE
Accept version strings with v

### LINKED ISSUE #79: Bad version strings make circup crash
```
❯ circup --path /Volumes/UMFS2/ list
Found device at /Volumes/UMFS2/, running CircuitPython 6.1.0.
A newer version of CircuitPython (6.2.0-beta.0) is available.
Traceback (most recent call last):
  File "circup.py", line 1054, in <module>
    main()
  File "/Library/Python/3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Python/3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Library/Python/3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Python/3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Python/3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Library/Python/3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "circup.py", line 881, in list
    modules = [m.row for m in find_modules(ctx.obj["DEVICE_PATH"]) if m.outofdate]
  File "circup.py", line 881, in <listcomp>
    modules = [m.row for m in find_modules(ctx.obj["DEVICE_PATH"]) if m.outofdate]
  File "circup.py", line 193, in row
    major_update = str(self.major_update)
  File "circup.py", line 174, in major_update
    == VersionInfo.parse(self.bundle_version).major
  File "/Users/spyro/Library/Python/3.8/lib/python/site-packages/semver.py", line 726, in parse
    raise ValueError("%s is not valid SemVer string" % version)
ValueError: v3.3.0 is not valid SemVer string
```
The culprint is `adafruit_si7021` whose bundle version string is `v3.3.0`.
Also looking at all the version strings pulled from the latest bundle, `adafruit_rplidar` has `0.0.1-auto.0` and makes circup crash too.
Every other modules follows the 1.2.3 format.

I guess fixing the tag in Adafruit_CircuitPython_SI7021 is the simplest way to fix that, but we may want to make circup more resilient to that kind of issue.
https://github.com/adafruit/Adafruit_CircuitPython_SI7021/tags

I wonder why adafruit_rplidar has a wrong version (latest release is 1.1.4)
https://github.com/adafruit/Adafruit_CircuitPython_RPLIDAR/tags

### PR DESCRIPTION
**Pull request by Neradoc**
Adds a "clean_version" function that removes the v in front of the version number so semver won't choke on it. There is a related idea that does a little more in the semver documentation:
https://python-semver.readthedocs.io/en/2.13.0/usage.html#dealing-with-invalid-versions

Also changes expecting a _`ValueError` in addition to a `TypeError`_ in `Module.major_update()`, so that bad version strings are caught. This fixes crashing circup. Remaining bad version numbers are assumed to be major updates.

This PR fixes #79 in regards to the adafruit_si7021 version. But adafruit_rplidar's `0.0.1-auto.0`, from which we can't extract the real version number anyway, will simply be assumed to need updating.

### CODE REVIEW

**Comment by Neradoc** [comment]
(Bear with me while I sort out running black, pylint, and make test locally before I push bad commits. Sorry.)

**Comment by tannewt** [comment]
I'd rather not support v####. Instead, can we make the deploys fail when a library is tagged with vsomething? Supporting two ways means that both will exist over the long term.

In CP we chose to omit `v` specifically because MicroPython versions start with `v`. By not using `v` we avoid tag collisions with upstream.

**Comment by Neradoc** [comment]
We can just fix circup crashing on it then, instead defaulting to treating invalid version tags as a major update, so that circup users are not blocked by it.

**Comment by makermelissa** [comment]
Yeah, maybe display a warning too, so that someone can be made aware of it.

**Comment by askpatrickw** [comment]
@Neradoc first off thanks for jumping in with a PR!
Are you cool morphing your PR to handle the validation and warnings mentioned?

**Comment by Neradoc** [comment]
I'm not sure how and what kind of warning would be relevant here to make the info come back to the maintainers.
"Bad version string, please consider filing an issue" with a  link to the module's repository ?

**Comment by makermelissa** [comment]
Yeah, that sounds good.

**Comment by askpatrickw** [comment]
@Neradoc  here's an example for a similar scenario

https://github.com/adafruit/circup/blob/master/circup.py#L675

**Comment by Neradoc** [comment]
I pushed a new version, it prints a warning in `circup update` with a link to the repo.
It also fixes `Module.repo` always being None (circup doesn't get `__repo__` from mpy files).

**Review by askpatrickw — APPROVED** [review]
I tested this locally, modifying a library on the device and in the bundle and it worked as expected.

Thanks @Neradoc

**Pull request merged**
